### PR TITLE
Add release notes for 1.4.0

### PR DESCRIPTION
**Pull request by trustin**
Based on #3316 - please ignore the dependency updates.

image

### CODE REVIEW

File: site/src/pages/release-notes/1.4.0.mdx [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+  // Getting
+  ContentDisposition disposition = headers.contentDisposition();
+  assert "attachment".equals(disposition.type());
+  assert "fileName".equals(disposition.name());
+  assert "file.png".equals(disposition.filename());
+
+  // Parsing
+  ContentDisposition parsed = ContentDisposition.parse(
+    "attachment; name=\"fileName\"; filename=\"file.png\");
```
Comment (minwoox) [inline site/src/pages/release-notes/1.4.0.mdx:126]:
```suggestion
    "attachment; name=\"fileName\"; filename=\"file.png\"");
```

File: site/src/pages/release-notes/1.4.0.mdx [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+  assert parsed.equals(disposition);
+  ```
+- You can now serve a file by returning an `HttpFile` in an annotated service. #3258 #3287
```
Comment (minwoox) [inline site/src/pages/release-notes/1.4.0.mdx:129]: `<type://HttpFile>`

File: site/src/pages/release-notes/1.4.0.mdx [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+              .expectComplete()
+              .verify()
+  
+  // You can also concat a `Publisher` of `Publisher`s.
+  StreamMessage<String> flat =
+    StreamMessage.concat(Flux.just(Mono.just("foo"),
+                                   Mono.just("bar")));
+  ```
+- You can now create an <type://HttpResponse> from a <type://ResponseHeaders> and
+  a stream of <type://HttpData>s using
```
Comment (minwoox) [inline site/src/pages/release-notes/1.4.0.mdx:171]: typeplural?

File: site/src/pages/release-notes/1.4.0.mdx [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+- Fixed the violation of Reactive Streams specification when `null` is published. #3212 #3216
+- <type://TomcatService> does not raise a `NullPointerException` when it stops.
+  #3136 #3218
+- Fixed stability and interoperability issues in Eureka service discovery module.
+  #3235 #3275 #3238 #3301
+- The service name generated from a CGLIB-enhanced class does not have synthetic
+  suffix anymore. #3103 #3240
+- <type://HealthCheckedEndpointGroup> now refreshes its <typeplural://Endpoint>'
+  status when their weights are changed. #3236
+- You'll no longer see a `NullPointerException` when retrying a gRPC request.
```
Comment (minwoox) [inline site/src/pages/release-notes/1.4.0.mdx:227]: missing #3251

**Review by minwoox — APPROVED** [review]
Huge work! Thanks a lot!

**Review by ikhoon — APPROVED** [review]
It was hard work. Thanks a lot @trustin! 🙇‍♂️

**Pull request merged**